python_modules/libraries/dagster-dg/dagster_dg/config.py

```diff
@@ -137,11 +140,96 @@ def _normalize_dg_partial_config(raw_dict: Mapping[str, object]) -> DgPartialCon
     if "is_deployment" in config and not isinstance(config["is_deployment"], bool):
         raise DgValidationError("`is_deployment` must be a boolean.")
 
+    if "code_locations" in config:
+        if not isinstance(config["code_locations"], list):
+            raise DgValidationError("`code_locations` must be a list.")
+        else:
+            normalized_code_location_specs = []
+            for spec in config["code_locations"]:
+                if not isinstance(spec, dict):
+                    raise DgValidationError("`code_locations` must be a list of dictionaries.")
+                normalized_code_location_specs.append(_normalize_code_location_spec(spec))
+        config["code_locations"] = normalized_code_location_specs
+
     if unrecognized_keys := [k for k in config.keys() if k not in DgPartialConfig.__annotations__]:
         raise DgValidationError(f"Unrecognized fields in configuration: {unrecognized_keys}")
     return cast(DgPartialConfig, config)
 
 
+def _normalize_code_location_spec(raw_dict: Mapping[str, object]) -> "CodeLocationSpec":
+    spec = {**raw_dict}  # copy so we can modify it
```

Comment:
I don't see why the copy is necessary

Reply:
There are modifications that (replacing str paths with Path objects for instance) that happen below. I expanded the comment.